src/content/content-events.js

```diff
@@ -46,15 +49,36 @@ export const onScriptLoaded = (timeout = 5000, interval = 100) => {
   });
 };
 
+const detectUrlChange = () => {
+  let previousUrl = '';
+  const observer = new MutationObserver(function () {
+    if (location.href !== previousUrl) {
+      previousUrl = location.href;
+      sendRealodMessageToWebapp();
+    }
+  });
+  observer.observe(document, { subtree: true, childList: true });
+};
```

Comment:
I liked this option more:
https://stackoverflow.com/a/47917728/839513

I don't want to observe any `document` mutation to know when the URL is changed.

Reply:
@rzarviv See my commit 947dbf2, I used the `history.psuh`

Reply:
@baruchiro I saw your commit, but I think I found a better solution for this issue - sending a message from `background.js` about URL change using `browser.tabs.onUpdated`. My solution doesn't require overriding the `history.pushState` event.

Reply:
Yes, please.